packages/rum-angular/test/specs/apm-service.spec.ts

```diff
@@ -0,0 +1,206 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import '../polyfills'
+import 'zone.js/dist/zone-testing'
+import { ApmService } from '../../src/apm.service'
+import { TestBed, ComponentFixture } from '@angular/core/testing'
+import { RouterTestingModule } from '@angular/router/testing'
+import {
+  BrowserDynamicTestingModule,
+  platformBrowserDynamicTesting
+} from '@angular/platform-browser-dynamic/testing'
+import { Component } from '@angular/core'
+import { Location } from '@angular/common'
+import { Routes, Router } from '@angular/router'
+import { apm } from '@elastic/apm-rum'
+
+@Component({
+  template: 'Home'
+})
+class HomeComponent {}
+
+@Component({
+  template: 'Slug'
+})
+class SlugComponent {}
+
+@Component({
+  template: '<router-outlet></router-outlet>'
+})
+class AppComponent {}
+
+const routes: Routes = [
+  { path: '', redirectTo: 'home', pathMatch: 'full' },
+  { path: 'home', component: HomeComponent },
+  { path: 'slug/:id', component: SlugComponent }
+]
+
+describe('ApmService', () => {
+  let service: ApmService
+  let router: Router
+  let location: Location
+  let fixture: ComponentFixture<AppComponent>
+  let compiled: HTMLElement
+
+  TestBed.initTestEnvironment(
+    BrowserDynamicTestingModule,
+    platformBrowserDynamicTesting()
+  )
+  ApmService.apm = apm
+
+  function setUpAppModule() {
+    TestBed.configureTestingModule({
+      imports: [RouterTestingModule.withRoutes(routes)],
+      declarations: [HomeComponent, AppComponent, SlugComponent],
+      providers: [ApmService]
+    })
+    service = TestBed.get(ApmService)
+    router = TestBed.get(Router)
+    location = TestBed.get(Location)
+    /**
+     * Inject router manually
+     */
+    service.router = router
+
+    fixture = TestBed.createComponent(AppComponent)
+    compiled = fixture.debugElement.nativeElement
+  }
+
+  function setUpRouteNavigation() {
+    fixture.ngZone.run(() => router.initialNavigation())
+  }
+
+  afterEach(() => {
+    apm.startTransaction.calls.reset()
```

Comment:
This won't be necessary once the previous comment is address, since we're starting with a new ApmBase instance!